test/runtests.jl

```diff
@@ -157,63 +160,104 @@ end
-
-        xcart = sph_to_cart(1, θ, ϕ)
-        vcart = [cos(ϕ)*cos(θ) -sin(ϕ)*sin(θ);
-                 sin(ϕ)*cos(θ)  cos(ϕ)*sin(θ);
-                 -sin(θ) 0] * v
-
-        for t=0:.1:1
-            @test isapprox(sph_to_cart(r, exp(M, x, v, t)...),
-                           r .* exp(S, xcart, vcart, t); atol=1e-3, rtol=1e-3)
-        end
 
-        Γ = christoffel_symbols_second(M, x)
-        for i=1:n
-            for j=1:n
-                for k=1:n
-                    if (i,j,k) == (2,2,1)
-                        @test Γ[i,j,k] ≈ -cos(θ)*sin(θ)
-                    elseif (i,j,k) == (2,1,2) || (i,j,k) == (1,2,2)
-                        @test Γ[i,j,k] ≈ cot(θ)
-                    else
-                        @test Γ[i,j,k] ≈ 0
+        for vtype in (Vector, SVector{n}, MVector{n})
+            x = vtype([θ, ϕ])
+            G = Diagonal(vtype([1, sin(θ)^2])) .* r^2
+            invG = Diagonal(vtype([1, 1 / sin(θ)^2])) ./ r^2
+            v, w = normalize(randn(n)), normalize(randn(n))
+
+            @test local_metric(M, x) ≈ G
+            @test inverse_local_metric(M, x) ≈ invG
+            @test det_local_metric(M, x) ≈ r^4 * sin(θ)^2
+            @test inner(M, x, v, w) ≈ dot(v, G * w)
+            @test norm(M, x, v) ≈ sqrt(dot(v, G * v))
+
+            xcart = sph_to_cart(1, θ, ϕ)
+            vcart = [cos(ϕ)*cos(θ) -sin(ϕ)*sin(θ);
+                     sin(ϕ)*cos(θ)  cos(ϕ)*sin(θ);
+                     -sin(θ) 0] * v
+
+            for t=0:.1:1
+                @test isapprox(sph_to_cart(r, exp(M, x, v, t)...),
+                               r .* exp(S, xcart, vcart, t); atol=1e-3, rtol=1e-3)
+            end
+
+            Γ₁ = christoffel_symbols_first(M, x)
+            for i=1:n
+                for j=1:n
+                    for k=1:n
+                        if (i,j,k) == (1,2,2) || (i,j,k) == (2,1,2)
+                            @test Γ₁[i,j,k] ≈ r^2*cos(θ)*sin(θ)
+                        elseif (i,j,k) == (2,2,1)
+                            @test Γ₁[i,j,k] ≈ -r^2*cos(θ)*sin(θ)
+                        else
+                            @test Γ₁[i,j,k] ≈ 0
+                        end
+                    end
+                end
+            end
+
+            Γ₂ = christoffel_symbols_second(M, x)
+            for i=1:n
+                for j=1:n
+                    for k=1:n
+                        if (i,j,k) == (2,2,1)
+                            @test Γ₂[i,j,k] ≈ -cos(θ)*sin(θ)
+                        elseif (i,j,k) == (1,2,2) || (i,j,k) == (2,1,2)
+                            @test Γ₂[i,j,k] ≈ cot(θ)
+                        else
+                            @test Γ₂[i,j,k] ≈ 0
+                        end
                     end
                 end
             end
+
+            R = riemann_tensor(M, x)
+            for i=1:n
+                for j=1:n
+                    for k=1:n
+                        for l=1:n
+                            if (i,j,k,l) == (1,1,2,2)
```

Comment:
this quadruple for-loop can be shortened to (and I was quite happy when I learned that)
````
for i=1:n, j=1:n, j=1:n, l=1:n
    [...]
end
```
see at the end of  https://docs.julialang.org/en/v1/manual/control-flow/#man-loops-1

Reply:
Oh that is nice. Thanks for the tip!